packages/synapse-react-client/src/components/dataaccess/UserAccessRequestHistory/UserAccessRequestHistoryTable.tsx

```diff
@@ -0,0 +1,181 @@
+import { upperFirst } from 'lodash-es'
+import { useMemo, useState } from 'react'
+import { Link } from 'react-router'
+import ColumnHeader from '../../TanStackTable/ColumnHeader'
+import StyledTanStackTable from '../../TanStackTable/StyledTanStackTable'
+import UserOrTeamBadge from '@/components/UserOrTeamBadge/UserOrTeamBadge'
+import { USER_ACCESS_HISTORY_SUBMISSION_SUBPATH } from './RouteConstants'
+
+const columnHelper = createColumnHelper<UserSubmissionSearchResult>()
+const columns = [
+  columnHelper.accessor('accessRequirementName', {
+    header: props => (
+      <ColumnHeader {...props} title={'Access Requirement Name'} />
+    ),
+    enableSorting: false,
+    enableColumnFilter: false,
+  }),
+  columnHelper.accessor('state', {
+    header: props => <ColumnHeader {...props} title={'Status'} />,
+    enableSorting: false,
+    cell: ctx => {
+      if (
+        ctx.row.original.userAccessApproval &&
+        ctx.row.original.userAccessApproval.expiredOn &&
+        dayjs(ctx.row.original.userAccessApproval.expiredOn).isBefore(dayjs())
+      ) {
+        return 'Expired'
+      }
+      return upperFirst(ctx.getValue()!.toLocaleLowerCase())
+    },
+    enableColumnFilter: false,
+    size: 75,
+  }),
+  columnHelper.accessor('createdOn', {
+    header: props => <ColumnHeader {...props} title={'Date Submitted'} />,
+    enableSorting: true,
+    cell: ctx => formatDate(dayjs(ctx.getValue())),
+    sortDescFirst: true,
+    enableColumnFilter: false,
+  }),
+  columnHelper.display({
+    id: 'expires',
+    header: props => <ColumnHeader {...props} title={'Expires'} />,
+    enableSorting: false,
+    cell: ctx => {
+      if (
+        ctx.row.original.userAccessApproval &&
+        ctx.row.original.userAccessApproval.expiredOn
+      ) {
+        return formatDate(dayjs(ctx.row.original.userAccessApproval.expiredOn))
+      }
+      return null
+    },
+    size: 75,
+    enableColumnFilter: false,
+  }),
+
+  columnHelper.accessor('submitterId', {
+    header: props => <ColumnHeader {...props} title={'Submitter'} />,
+    enableSorting: false,
+    cell: ctx => <UserOrTeamBadge principalId={ctx.getValue()} />,
+    enableColumnFilter: false,
+  }),
+
+  columnHelper.accessor('id', {
+    header: props => <ColumnHeader {...props} title={''} />,
+    enableSorting: false,
+    cell: ctx => (
+      <>
+        <Link
+          to={`/${USER_ACCESS_HISTORY_SUBMISSION_SUBPATH}/${ctx.getValue()}`}
+        >
+          View Request
+        </Link>
+      </>
+    ),
+    size: 95,
+    enableColumnFilter: false,
+  }),
+]
+
+function getSortApiRequestFromTableSortState(
+  sortingState: SortingState,
+): SubmissionSearchSort[] | undefined {
+  if (sortingState.length === 0) {
+    return undefined
+  }
+  const sort = sortingState[0]
+  let field: SubmissionSortField = SubmissionSortField.CREATED_ON
+  if (sort.id === 'createdOn') {
+    field = SubmissionSortField.CREATED_ON
+  } else if (sort.id === 'modifiedOn') {
```

Comment:
This is fine, but is modifiedOn actually shown in the table (and a sortable tanstack header)?

Reply:
It's not -- this was copied from the ACT table (the search functionality uses the same Java model) but the ACT version uses our hand-defined types whereas this one uses the OpenAPI types. I'll consolidate in the next PR.